bundler/lib/bundler/cli.rb

```diff
@@ -292,6 +292,8 @@ def install
       "Prefer updating only to next minor version"
     method_option "major", :type => :boolean, :banner =>
       "Prefer updating to next major version (default)"
+    method_option "pre", :type => :boolean, :banner =>
+      "If updating, allow consideration of prerelease gems"
```

Comment:
Prerelease gems are already considered when dependency specifications mention prereleases, for example if your `Gemfile` or a transitive gemspec specifies something like `rails >= 7.0.0.beta1`, or when they are the only possibility.

Also, the `If updating` seems redundant in the case of `bundle update`, right?

How could we word this flag's help message to clarify the above?

Maybe

```suggestion
      "Always choose the highest allowed version when updating gems, even if it's a prerelease"
```

Reply:
:laughing: no problem! Totally with you on this needing clarification.

What I think is important to convey is that prerelease gems are typically (outside the case you mention) deprioritized and that this flag allows avoiding that. Avoiding deprioritization is a nice double negative that I'm having trouble avoiding, but tweaking your suggestion slightly to `Always chose the highest allowed version when updating gems, regardless of prerelease status` is closer.

What do you think about `"Prioritize versions for selection independent of whether they're prerelease"`? Not loving it (especially prioritize versions for selection seems perhaps too low level for here, but it seems closest to what I imagine is important to convey. Otherwise, happy to go with the suggested text!

Reply:
My rationale of mentioning "highest allowed version" is that without being clear on how things are ordered, talking about "prioritizing" feels a bit meaningless.

So from your two suggestions, I'd go with `Always chose the highest allowed version when updating gems, regardless of prerelease status`. I'm neutral between that one, my suggestion above, or some other variant like `Always chose the highest allowed version when updating gems, regardless of whether it's a prerelease`.

So just update with your preferred wording along those lines, and let's ship this!

Reply:
Sounds good! I kept the `if updating` in the lock version and removed the `when updating gems` since all the other similar descriptions in `lock` started that way, but let me know if you'd prefer consistency between the two `--pre` descriptions.